sambacc/ctdb.py

```diff
@@ -704,6 +704,42 @@ def check_nodestatus(cmd: samba_cmds.SambaCommand = samba_cmds.ctdb) -> None:
     samba_cmds.execute(cmd_ctdb_check)
 
 
+def _read_command_pnn(cmd: samba_cmds.SambaCommand) -> typing.Optional[int]:
+    """Run a ctdb command assuming it returns a pnn value. Return the pnn as an
+    int on success, None on command failure.
+    """
+    try:
+        out = subprocess.check_output(list(cmd))
+        pnntxt = out.decode("utf8").strip()
```

Comment:
I think this could yield `UnicodeDecodeError`. Should be handled with proper `except`.

Reply:
It's extremely unlikely given the output of the ctdb command. I'd prefer the let the exception bubble up to the top level if it occurs.

Reply:
Fair enough